api/secrets/initialize.js

```diff
@@ -247,62 +255,237 @@ function _initializeVault(bag, next) {
   );
 }
 
+function _checkInitStatus(bag, next) {
+  var who = bag.who + '|' + _checkInitStatus.name;
+  logger.verbose(who, 'Inside');
+
+  var client = new VaultAdapter(bag.vaultUrl);
+
+  client.getInitializedStatus(
+    function (err, response) {
+      if (err)
+        return next(
+          new ActErr(who, ActErr.OperationFailed,
+            'Failed to get vault initialized status: ' + util.inspect(err))
+        );
+
+      if (response.initialized === true) {
+        logger.debug('Vault already initialized');
+
+        if (_.isEmpty(bag.config.unsealKey1) ||
+          _.isEmpty(bag.config.unsealKey2) ||
+            _.isEmpty(bag.config.unsealKey3) ||
+              _.isEmpty(bag.config.unsealKey4) ||
+                _.isEmpty(bag.config.unsealKey5) ||
+                  _.isEmpty(bag.config.rootToken)) {
+
+          return next(
+            new ActErr(who, ActErr.OperationFailed,
+              'Vault needs 5 unseal keys. One(or more) unseal keys are ' +
+              ' missing. Reset vault server or make sure keys are present')
+          );
+        }
+        bag.config.isInitialized = response.initialized;
+      } else {
+        bag.config.isInitialized = false;
+      }
+      return next();
+    }
+  );
+}
+
 function _getUnsealKeys(bag, next) {
+  if (bag.config.isInitialized) return next();
+
   var who = bag.who + '|' + _getUnsealKeys.name;
   logger.verbose(who, 'Inside');
 
-  var keyIndex = 1;
-  var unsealKeysFile = path.join(
-    global.config.configDir, bag.component, 'scripts/keys.txt');
+  var client = new VaultAdapter(bag.vaultUrl);
+  var params = {
+    secret_shares: 5,
+    secret_threshold: 3
+  };
+  client.init(params,
+    function (err, response) {
+      if (err)
+        return next(
+          new ActErr(who, ActErr.OperationFailed,
+            'Failed to initialize vault: ' + util.inspect(err))
+        );
+      bag.config.unsealKey1 = response.keys[0];
+      bag.config.unsealKey2 = response.keys[1];
+      bag.config.unsealKey3 = response.keys[2];
+      bag.config.unsealKey4 = response.keys[3];
+      bag.config.unsealKey5 = response.keys[4];
+      bag.config.rootToken  = response.root_token;
+
+      logger.debug('Successfully fetched unseal keys for vault');
+      return next();
+    }
+  );
+}
 
-  var filereader = readline.createInterface({
-    input: fs.createReadStream(unsealKeysFile),
-    console: false
-  });
+function _saveUnsealKeys(bag, next) {
```

Comment:
can you skip this if it's already initialized? If it were initialized, it ooks like you wouldn't have any changes to the config object to put.